--- a/make/common/native/CompileFile.gmk	2026-03-10 00:54:39.983332783 +0100
+++ b/make/common/native/CompileFile.gmk	2026-03-10 00:54:40.031733333 +0100
@@ -130,13 +130,13 @@ define CreateCompiledNativeFileBody
       # Compile as a C file
       $1_CFLAGS += $$($1_WARNINGS_FLAGS)
       $1_FLAGS := $(CFLAGS_CCACHE) $$($1_USE_PCH_FLAGS) $$($1_BASE_CFLAGS) \
-          $$($1_OPT_CFLAGS) $$($1_CFLAGS) -c
+          $$($1_CFLAGS) -c
       $1_COMPILER := $$($$($1_BASE)_CC)
     else ifneq ($$(filter %.m, $$($1_FILENAME)), )
       # Compile as an Objective-C file
       $1_CFLAGS += $$($1_WARNINGS_FLAGS)
       $1_FLAGS := -x objective-c $(CFLAGS_CCACHE) $$($1_USE_PCH_FLAGS) \
-          $$($1_BASE_CFLAGS) $$($1_OPT_CFLAGS) $$($1_CFLAGS) -c
+          $$($1_BASE_CFLAGS) $$($1_CFLAGS) -c
       $1_COMPILER := $$($$($1_BASE)_CC)
     else ifneq ($$(filter %.S, $$($1_FILENAME)), )
       # Compile as preprocessed assembler file
@@ -160,7 +160,7 @@ define CreateCompiledNativeFileBody
       # Compile as a C++ or Objective-C++ file
       $1_CXXFLAGS += $$($1_WARNINGS_FLAGS)
       $1_FLAGS := $(CFLAGS_CCACHE) $$($1_USE_PCH_FLAGS) $$($1_BASE_CXXFLAGS) \
-          $$($1_OPT_CXXFLAGS) $$($1_CXXFLAGS) -c
+          $$($1_CXXFLAGS) -c
       $1_COMPILER := $$($$($1_BASE)_CXX)
     else
       $$(error Internal error in NativeCompilation.gmk: no compiler for file $$($1_FILENAME))
@@ -290,7 +290,7 @@ define CreatePrecompiledHeader
         -include $$($1_PCH_DEPS_TARGETS_FILE)
 
         $1_PCH_COMMAND := $$($1_CC) $$($1_CFLAGS) $$($1_EXTRA_CFLAGS) $$($1_SYSROOT_CFLAGS) \
-            $$($1_OPT_CFLAGS) -x c++-header -c $(GENDEPS_FLAGS) \
+            -x c++-header -c $(GENDEPS_FLAGS) \
             $$(addsuffix .tmp, $$($1_PCH_DEPS_FILE))
 
         $$($1_PCH_FILE): $$($1_PRECOMPILED_HEADER) $$($1_COMPILE_VARDEPS_FILE)
